src/Extractor/BaseExtractor.php

```diff
@@ -50,44 +40,25 @@ public function __construct(array $parameters, array $state, LoggerInterface $lo
         $this->dataDir = $parameters['data_dir'];
         $this->state = $state;
         $this->logger = $logger;
-        if (isset($parameters['db']['ssh']['enabled']) && $parameters['db']['ssh']['enabled']) {
-            try {
-                $sshTunnel = new SSHTunnel($logger);
-                $parameters['db'] = $sshTunnel->createSshTunnel($parameters['db']);
-            } catch (SSHTunnelUserException $e) {
-                throw new UserException($e->getMessage(), 0, $e);
-            }
-        }
-        $this->databaseConfig = $this->createDatabaseConfig($parameters['db']);
-
-        $proxy = new DbRetryProxy($this->logger, self::CONNECT_MAX_RETRIES, [PDOException::class]);
 
-        try {
```

Comment:
Vytvaranie spojenia je presunute to metody `BaseExtractor::connect`
- https://github.com/keboola/db-extractor-common/blob/e1196ba07da0cf526ac4468ca7a45a18019245a7/src/Extractor/BaseExtractor.php#L145-L152
- Exception` Missing dirver: ...` bola odstranena, pretoze je to pre-historicky pozostatok z doby, ked jeden extractor pouzival viacero PDO ovladacov. Kazdy extractor obsahuje testy, takze ak by chybal driver, tak spadnu.
- Z tohto miesta bolo odstranene retry a presunute do: `BaseDbConnection::connectWithRetry`
https://github.com/keboola/db-extractor-adapter/blob/c12f04421c0d73cde5a0bb3ef94547affc26e72e/src/Connection/BaseDbConnection.php#L102-L113
- Kedze to v tychto novych `*DbConnection` triedach dava vacsi zmysel, ... a moze sa to pre rozne spojenia podla potreby upravit.